packages/notebook/src/widget.ts

```diff
@@ -3416,6 +3435,27 @@ export class Notebook extends StaticNotebook {
     }
   }
 
+  /**
+   * Handle changes to the notebook model.
+   */
+  private _onNotebookModelChanged(): void {
+    if (this.model) {
+      const cells = this.model.cells;
+      for (let i = 0; i < cells.length; i++) {
+        cells.get(i).contentChanged.connect(this._onCellContentChanged, this);
+      }
+    }
+  }
+
+  /**
+   * Handle a cell content change.
+   */
+  private _onCellContentChanged(sender: ICellModel): void {
+    const cell = this.widgets.find(w => w.model === sender) || null;
+    this._lastModifiedCell = cell;
```

Comment:
Maybe we could store a stack of cells so that this command can be run multiple times, going back in history of modified cells (like jumping a few cells back). That is a bit harder problem to tackle, but if we are making additions to the public API (`lastModifiedCell`) then it's worth giving it more thought now as it will be harder to change later.